Armadillo/src/main/java/com/scribd/armadillo/playback/mediasource/DashMediaSourceGenerator.kt

```diff
@@ -14,31 +13,23 @@ import javax.inject.Inject
 
 internal class DashMediaSourceGenerator @Inject constructor(
     private val mediaSourceHelper: HeadersMediaSourceHelper,
-    private val downloadTracker: DownloadTracker) : MediaSourceGenerator {
+    private val downloadTracker: DownloadTracker,
+    private val drmMediaSourceHelper: DrmMediaSourceHelper,
+) : MediaSourceGenerator {
 
     override fun generateMediaSource(context: Context, request: AudioPlayable.MediaRequest): MediaSource {
         val dataSourceFactory = mediaSourceHelper.createDataSourceFactory(context, request)
 
         downloadTracker.getDownload(request.url.toUri())?.let {
             if (it.state != Download.STATE_FAILED) {
-                return DownloadHelper.createMediaSource(it.request, dataSourceFactory)
+                val mediaItem = drmMediaSourceHelper.createMediaItem(context = context, request = request, isDownload = true)
```

Comment:
Does this need to get bypassed for non-DRM content?

Reply:
In `DrmMediaSourceHelper`, we check if the DRM object is non-null before adding anything DRM-related, so we're okay with not checking it here